src/main/java/com/pszymczyk/Reservation.java

```diff
@@ -2,49 +2,45 @@
 
 import java.util.UUID;
 
-public class Reservation {
+class Reservation {
 
-    public enum ReservationStatus {
+    enum ReservationStatus {
         NEW,
         CONFIRMED,
         CANCELED
-
     }
 
-    private UUID id;
-    private String userId;
-    private String tripCode;
-    private ReservationStatus status;
+    private final UUID id;
+    private final String userId;
+    private final ReservationStatus status;
 
-    public UUID getId() {
-        return id;
+    Reservation(String userId) {
+        this.id = UUID.randomUUID();
+        this.userId = userId;
+        this.status = ReservationStatus.NEW;
     }
 
-    public void setId(UUID id) {
+    Reservation(UUID id, String userId, ReservationStatus status) {
         this.id = id;
-    }
-
-    public String getUserId() {
-        return userId;
-    }
-
-    public void setUserId(String userId) {
         this.userId = userId;
+        this.status = status;
     }
 
-    public String getTripCode() {
-        return tripCode;
-    }
-
-    public void setTripCode(String tripCode) {
-        this.tripCode = tripCode;
+    Reservation cancel() {
+        return new Reservation(
```

Comment:
metoda sugeruje ze zostanie tylko zmienione pole status, a dostajemy nowy obiekt

Reply:
Byłem kiedyś na szkoleniu i tam mówili, że trzeba robić niemutowalne. Ale rzeczywiście skoro id sie nie zmienia to bardziej logiczne będzie modyfikowanie stanu obiektu.